bfd/BFD_Enhancement_HLD.md

```diff
@@ -0,0 +1,421 @@
+
+Supporting ECMP is challenging in BFD due to time it may take for control plane to know that the active nexthop went down. In SONiC this information has to traverse all the way down to kernel after traversing all the DBs this may take considerable time.                                                                         
+
+In current SONiC implementation BFD relies on kernel network stack to switch the BFD packet to next available nexthop when a active nexthop goes down. BFD timers in this case is directly proportional to the time it takes for kernel to get the nexthop down event. Faster the kernel learns active nexthop is down the more aggressive BFD timers can be. In this case it is suggested to configure BFD timer values to have a timeout of atleast 600 msec.                                                                                                                                            
+
+## 3.2 CLI
+### 3.2.1 Data Models
+NA                   
+### 3.2.2 Configuration Commands
+
+Config commands as  in FRR BGP container is described at below link.  
+[Config commands](http://docs.frrouting.org/en/latest/bfd.html)
+### 3.2.3 Show Commands
+
+Show commands as  in FRR BGP container is described at below link.  
+[Show commands](http://docs.frrouting.org/en/latest/bfd.html)  
+**show bfd peer <WORD label | <A.B.C.D|X:X::X:X> [{multihop|local-address <A.B.C.D|X:X::X:X>|interface IFNAME ifname|vrf NAME vrfname}]**
+
+This command is enhanced to add detect-multiplier, enhanced show output is as below:
+```  
+sonic# show bfd peer
+    BFD Peers:          
+        peer 10.0.0.103 interface Ethernet204
+            ID: 2                            
+            Remote ID: 5                     
+            Status: up                       
+            Uptime: 30 second(s)             
+            Diagnostics: ok                  
+            Remote diagnostics: ok           
+            Local timers:  
+                Detect-multiplier: 3                  
+                Receive interval: 300ms      
+                Transmission interval: 300ms 
+                Echo transmission interval: disabled
+            Remote timers:
+                Detect-multiplier: 3 
+                Receive interval: 200ms
+                Transmission interval: 200ms
+                Echo transmission interval: 0ms                                                                                              
+```  
+
+A new show command is added as below:
+
+**show bfd peers brief**
+Show all the BFD peer in brief, sample output as below:
+This command is available in FRR vtysh shell.
+
+``` 
+Session count: 1
+OurAddr                  NeighAddr                        State    
```

Comment:
Pls change OurAddr to LocalAddr maybe.
Can you include a column for BFD session Id as well

Reply:
Will update